tests/performance-test/README.md

```diff
@@ -3,18 +3,139 @@
 ## Introduction
 
 The performance test provides an automated environment in which to to run stress
-tests on the SAF locally using Minishift. Collectd-tg is used to simulate
-extensive netwrok traffic to pump through SAF. Because Minishift only supports a
-single node at a time, this test demonstrates the limits of SAF in a constrained
-environment. Test scenarios are manually configured in a yaml file and results
-can be analyzed in a series of grafana dashboards.
+tests on SAF. Collectd-tg or telemetry-bench are used to simulate extensive
+metrics data to pump through SAF. Results of testing can be analyzed in a
+grafana dashboard.
 
 Two additional pods are deployed by the performance test: one that hosts a
 grafana instance and one that executes the testing logic.
 
 ![A Performance Test Dashboard](images/dashboard.png)
 
-## Configuring Tests
+## Environment
+
+* openshift v3.11.135
+* docker v17.05+
+
+## Setup
+
+SAF must already be deployed, a quick way to do this is using the
+`quickstart.sh` script in `telemetry-framework/deploy/` directory to run SAF.
+Here is an example of how to do that in minishift:
+
+>NOTE: if you intend to rebuild any test containers, your openshift must have an
+>external route for the registry. The command for that is included here.
+
+```shell
+minishift addons enable registry-route   # Run BEFORE starting minishift
+minishift start
+eval $(minishift oc-env)
+cd $WORKDIR/telemetry-framework/deploy/; ./quickstart.sh
+```
+
+More details about deploying SAF can be found in the
+[SAF deployment docs](../../../docs/README.adoc#quickstart-script-for-development-usage)
+
+## Deploying Grafana
+
+Ensure that all of the SAF pods are already marked running with `oc get pods`.
+Next, launch the grafana instance for test results gathering. This only needs
+to be done once:
+
+```shell
+cd $WORKDIR/telemetry-framework/tests/performance-test/deploy
+./grafana-launcher.sh
+```
+
+The grafana launcher script will output a URL that can be used to log into the
```

Comment:
I was trying to think if maybe it would be worth noting that this instance of Grafana is not for the purposes of production deployments. While this is mostly for upstream development and testing, at this point we do technically ship all of the testing platform stuff since we only release tags/releases from the upstream repo, and not from downstream at all.

It's possible we should change that at some point, and create a `telemetry-framework` or `service-assurance` repo downstream that contains the same contents as upstream, but perhaps with the performance testing framework stripped out?

Not sure I have a requested change here. More just thinking out loud. Feel free to add something if you think it's worth adding.

Reply:
This documentation is under the tests/performance-test/README.md - I would hope it wouldn't be confused with a production-ready grafana.

That said, I'm pretty sure we decided that we do actually "want" to ship Grafana anyways (I'm already excited to spend way too much time on the minutiae of an OSP monitoring dashboard!) at which point this would just go away I guess.

Also just musing; I don't think anything needs to change here.

Reply:
Sounds good! For the upstream Grafana we'll want to look at using https://github.com/integr8ly/grafana-operator as I believe that is the closest operator we have to being made into a downstream component.

Reply:
Let's do that after we have a top-level SAF operator

Reply:
I think I'm going to have to put away whatever outstanding Events work we have, and just start on that. I have a meeting to start getting that work prep'd with @aneeshkp tomorrow so that I can tackle that epic over our next sprint.